protop-core/src/main/java/io/protop/core/sync/SyncService.java

```diff
@@ -157,25 +196,33 @@ private void filterValidChildren(DirectoryWithFiles directoryWithFiles, File pat
     /**
      * Create subdirectories and symlink files in the given parent.
      */
-    private void mergeChildrenToParentDirectory(File parent, DirectoryWithFiles children) throws IOException {
-        for (Map.Entry<String, File> fileEntry : children.getFiles().entrySet()) {
-            Files.createSymbolicLink(parent.toPath().resolve(fileEntry.getKey()),
-                    fileEntry.getValue().toPath());
+    private void mergeChildrenToParentDirectory(
+            final Path protopPathDir, final Path parent, final DirectoryWithFiles children) throws IOException {
+        for (Map.Entry<String, FileWithRootDir> fileEntry : children.getFiles().entrySet()) {
+            final Path link = parent.resolve(fileEntry.getKey());
+
+            fileEntry.getValue().createSymbolicLink(protopPathDir, link);
```

Comment:
Won't this keep replacing the target? I'm not sure this will work with deeply nested structures at all, but I will test it to verify.

Reply:
Do you mean on successive syncs? I'll write a test for this scenario.

Reply:
Oh, I see what you mean. The function is recursive. Lemme think what can and should be done here.